tests/conftest.py

```diff
@@ -129,11 +129,8 @@ def tmpdir_factory(request, tmpdir_factory):
     """
     yield tmpdir_factory
     if not request.config.getoption("--keep-tmpdir"):
-        # py.path.remove() uses str paths on Python 2 and cannot
-        # handle non-ASCII file names. This works around the problem by
-        # passing a unicode object to rmtree().
         shutil.rmtree(
-            str(tmpdir_factory.getbasetemp()),
+            tmpdir_factory.getbasetemp(),
             ignore_errors=True,
         )
```

Comment:
Can this one not be `tmpdir_factory.getbasetemp().remove(...)`?

Reply:
`pathlib.Path.remove` isn't a thing. :P

`pathlib.Path.unlink` expects it to be a file or empty directory: https://docs.python.org/3/library/pathlib.html#pathlib.Path.unlink

Reply:
That's failing on Windows:
```
        if not request.config.getoption("--keep-tmpdir"):
>           tmpdir_factory.getbasetemp().remove(ignore_errors=True)
E           py.error.ENOENT: [No such file or directory]: listdir('R:\\Temp\\pytest-of-VssAdministrator\\pytest-0\\popen-gw1\\test_absolute_http_nested_req_current',)
```
https://dev.azure.com/pypa/pip/_build/results?buildId=30820&view=logs&jobId=f801bdd4-2ff4-5da8-3bd6-deea013b4470&j=f801bdd4-2ff4-5da8-3bd6-deea013b4470&t=16f6b271-31b4-5cd6-9f4b-0d32a26b2e2b

Reply:
I've re-run the job, in case it's some sort of flaky pytest behaviour.

Reply:
> `pathlib.Path.remove` isn't a thing. :P

But `tmpdir_factory.getbasetemp()` returns `py.path.LocalPath`, not `pathlib.Path` :p

> That's failing on Windows

Probably need to do somwthing like

```python
path = tmpdir_factory.getbasetemp()
if path.check(exists=True):
    path.remove(ignore_errors=True, rec=True)
```

Reply:
`rec=1` by default, so could skip that param: https://github.com/pytest-dev/py/blob/25e537fa5024208647aa121e371b24f094008c31/py/_path/local.py#L207

Maybe the stdlib `shutil` version would be cleaner than a couple of calls to the third-party `py`, which ends up calling `shutil.rmtree` anyway?

```python
shutil.rmtree(
    tmpdir_factory.getbasetemp(),
    ignore_errors=True,
)
```

Reply:
I agree -- the `shutil.rmtree` form is cleaner than the `py._path.local` format. :)

Reply:
Updated and reverted to `shutil.rmtree`